src/user_management/user.model.js

```diff
@@ -1,11 +1,40 @@
+import bcrypt from 'bcrypt'
+const saltRounds = 10
+
 export default (sequelize, DataTypes) => {
   const User = sequelize.define('user', {
     id: {
       type: DataTypes.UUID,
       primaryKey: true,
       defaultValue: DataTypes.UUIDV4
+    },
+    firstName: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    lastName: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    email: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    password: {
+      type: DataTypes.STRING,
+      allowNull: false
     }
   })
 
+  User.prototype.validatePassword = function (password) {
+    return bcrypt.compare(password, this.password)
+  }
+
+  User.beforeCreate(async function (password) {
+    const salt = await bcrypt.genSalt(saltRounds)
+    const hashedPassword = await bcrypt.hash(password, salt)
+    this.password = hashedPassword
+  })
```

Comment:
You also need to do this when an `update` happens. *Don't copy/paste the code*. Extract the password hashing to an instance method and then use that method on both hooks (`beforeCreate` and `beforeUpdate`).

Reply:
Also, you need to extract the password from `this.password`, not received as a parameter.